src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -150,12 +156,19 @@ public List<DBTableColumn> searchColumnsByTag(String tag) {
 			sourceColumns = sourceColumnRepository.retrieveByTag(tag.toLowerCase());
 
 		}
+		if (sourceColumns.isEmpty()) {
+			throw new ResourceNotFoundException("No tables found with the given tag.");
```

Comment:
It should be No Content. 
ResponseStatusException(HttpStatus.NO_CONTENT

Reply:
resolved